lib/Sema/CSDiag.cpp

```diff
@@ -2617,7 +2618,7 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
         // Fetch any declaration to check if the name is '~='
         ValueDecl *decl0 = overloadedFn->getDecls()[0];
 
-        if (decl0->getName() == decl0->getASTContext().Id_MatchOperator) {
+        if (decl0->getBaseName() == decl0->getASTContext().Id_MatchOperator) {
```

Comment:
I never wanted to make `self` a special name (neither do I think it should be one). The above was just an explanation on why I chose to compare to `Id_self` using `getFullName()` (thus taking the `==` operator of `DeclName)` and compare to any other identifier (e.g. `Id_MatchOperator`) using `getBaseName()` (taking the `==` operator of `DeclBaseName`).